pkg/store/sources_test.go

```diff
@@ -5,10 +5,122 @@ Copyright 2020 Adevinta
 package store
 
 import (
+	"context"
+	"database/sql/driver"
 	"reflect"
 	"testing"
+	"time"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/jmoiron/sqlx"
+)
+
+var (
+	tTime   = time.Now()
+	tSource = Source{
+		ID:       "id_source_1",
+		Instance: "id_check_1",
+		SourceFamily: SourceFamily{
+			Name:      "family_1",
+			Component: "vulcan",
+			Target:    "api.example.com",
+		},
+		Time: tTime,
+	}
+	createSourceQueryArgs = []driver.Value{
+		tSource.Name,
+		tSource.Component,
+		tSource.Instance,
+		tSource.Options,
+		tSource.Time,
+		tSource.Target,
+	}
 )
 
+func TestCreateSource(t *testing.T) {
+	tests := []struct {
+		name               string
+		insertRowsResponse *sqlmock.Rows
+		selectRowsResponse *sqlmock.Rows
+		expectErr          bool
+		inputSource        *Source
+		expectedSource     *Source
+	}{
+		{
+			name: "CreateNonExistingSource",
+			insertRowsResponse: sqlmock.NewRows(
+				[]string{"id", "name", "component", "instance", "options", "time", "target_id"},
+			).AddRow(
+				"id_source_1", "family_1", "vulcan", "id_check_1", "", tTime, "api.example.com",
+			),
+			expectErr:      false,
+			inputSource:    &tSource,
+			expectedSource: &tSource,
+		},
+		{
+			name:               "CreateExistingSource",
+			insertRowsResponse: sqlmock.NewRows([]string{}),
+			selectRowsResponse: sqlmock.NewRows(
+				[]string{"id", "name", "componet", "instance", "options", "time", "target_id"},
+			).AddRow(
+				tSource.ID, tSource.Name, tSource.Component, tSource.Instance, tSource.Options, tSource.Time, tSource.Target,
+			),
+			expectErr:      false,
+			inputSource:    &tSource,
+			expectedSource: &tSource,
+		},
+		{
+			name:               "CreateMalformedNonExistingSource",
+			insertRowsResponse: sqlmock.NewRows([]string{}),
+			expectErr:          true,
+			inputSource:        &Source{},
+			expectedSource:     nil,
+		},
+		{
+			name:               "CreateMalformedExistingSource",
+			insertRowsResponse: sqlmock.NewRows([]string{}),
+			selectRowsResponse: sqlmock.NewRows([]string{}),
+			expectErr:          true,
+			inputSource:        &tSource,
+			expectedSource:     nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockDB, mock, _ := sqlmock.New()
+			defer mockDB.Close()
+			sqlxDB := sqlx.NewDb(mockDB, "sqlmock")
+
+			mock.ExpectBegin()
+			mock.ExpectQuery("INSERT INTO").
+				WithArgs(createSourceQueryArgs...).
+				WillReturnRows(tt.insertRowsResponse)
+
+			if tt.selectRowsResponse != nil {
+				mock.ExpectQuery("^SELECT (.+) FROM sources").
+					WithArgs(createSourceQueryArgs...).
+					WillReturnRows(tt.selectRowsResponse)
+			}
+			ctx := context.Background()
+			tx, _ := sqlxDB.BeginTxx(ctx, nil)
+			screated, err := createSource(ctx, tx, *tt.inputSource)
+			if tt.expectErr && err == nil {
+				t.Errorf("an error was expected creating source: %s in test: %s", err, tt.name)
```

Comment:
I believe here we do not need to use the `tt.name` in the error message as we are already using `t.Run` passing the test name, so I think that will be displayed by Go in any case.